include/nbl/system/ISystem.h

```diff
@@ -61,10 +65,11 @@ class NBL_API2 ISystem : public core::IReferenceCounted
         {
             E_REQUEST_TYPE type;
             std::variant<
```

Comment:
any reason why you need a type because std::variant provides something of the sort.

Reply:
just for this: https://github.com/Devsh-Graphics-Programming/Nabla/blob/61fa8cdedd3868f8e22c8790874eddcf72e96d57/src/nbl/system/ISystem.cpp#L311

Reply:
I know that you can try to resolve the variant for each type i mean.

https://en.cppreference.com/w/cpp/utility/variant/get_if